edx_solutions_api_integration/users/views.py

```diff
@@ -459,6 +459,32 @@ def post(self, request):  # pylint: disable=R0915
         return Response(response_data, status=status.HTTP_201_CREATED)
 
 
+class TokenBasedUserDetails(TokenBasedAPIView):
+    """
+    ### The TokenBasedUserDetails view allows clients to interact with a specific User entity
+    - URI: ```/api/server/users/validate-token```
+    - GET: Returns a JSON representation of the specified User entity
+
+    ### Use Cases/Notes:
+    * Use the TokenBasedUserDetails view to obtain the current state for a specific User
+      by providing Bearer token in request
+    """
+
+    def get(self, request):
+        """
+        GET /api/server/users/validate-token
+        """
+        response_data = {}
+        try:
+            existing_user = User.objects.get(id=request.user.id)
```

Comment:
do we need this query to fetch user object. don't  we have user object already?
@shafqatfarhan